apps/api/src/app/shared/dtos/preference-channels.ts

```diff
@@ -1,14 +1,29 @@
 import { ApiPropertyOptional } from '@nestjs/swagger';
+import { IsBoolean, IsOptional } from 'class-validator';
```

Comment:
I decided to reuse this dto for validation, because it's possible to create completely corrupted `preferenceSettings` in database (in update at least).